src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -51,6 +51,13 @@ module sonic-port{
 						length 1..128;
 					}
 				}
+                
+				leaf mode {
+					description "SwitchPort Modes values will be from "routed|access|trunk. Default val is routed. "; 
+					 type string {
+						length 1..255;	
+					}
```

Comment:
Why not use pattern here?
For example, pattern "routed|access|trunk"

Reply:
yes , pattern can also be used. Will update it accordingly

Reply:
modified code as per suggestions